orion-server/src/main/java/com/pinterest/orion/core/actions/kafka/BrokerRecoveryAction.java

```diff
@@ -207,7 +208,15 @@ protected boolean checkServiceStabilityAfterRestart(String nodeId) {
 
   @Override
   public String getName() {
-    return "Broker Recovery" + (isDryRun ? " - Dry Run" : "");
+    // Different action names are required for BrokerRecoveryAction to be dispatched from same ClusterRecoveryAction.
+    String nodeId = "Unknown Node";
+    if (containsAttribute(OrionConstants.NODE_ID)) {
+      nodeId = getAttribute(OrionConstants.NODE_ID).getValue();
+    }
```

Comment:
Is it possible for the if condition to fail for two nodes? Would it be an issue if two `nodeId`s are the same (`Unknown Node`)?

Reply:
Action being created without node id is an edge case with issue. Unable to create that action is not an issue. I can make uuid here but I think it's overkill.